portfolio/src/main/webapp/script.js

```diff
@@ -76,6 +75,27 @@ function getDatastoreComments(){
   });
 }
 
+function showComments() {
+  fetch('/login').then(response => response.json()).then((status) => {
+    const headerContainer = document.getElementById('header');
+    if (headerContainer.childElementCount == 3) {
```

Comment:
Why the check against 3 here? This deserves at least a comment, or if possible, a change to make the code itself describe the intent more explicitly.